synfig-studio/src/gui/dialogs/canvasresize.cpp

```diff
@@ -259,6 +262,27 @@ void CanvasResize::on_height_changed()
 	refresh_wh_toggle_widgets();
 }
 
+void CanvasResize::on_size_template_changed(){
+	String selection(combo_box->get_active_id());
+	if(selection == "custom") {
+		width->set_sensitive(true);
+		height->set_sensitive(true);
+		return;
+	}
+	String::size_type locx=selection.find_first_of('x');
+	String::size_type strSize=selection.size();
+	String x_size(selection.substr(0,locx));
```

Comment:
Just for safety, check if `locx` isn't `std::string::npos` (i.e., if character 'x' was found).